tests-upgrade/tests-sdk1-support/AutoRestSupportSdkTest.ps1

```diff
@@ -0,0 +1,262 @@
+        [string]
+        $CsharpFilePath,
+
+        [string]
+        $PowerShellFilePath
+    )
+    $fileName = Split-Path -Path $CsharpFilePath -Leaf
+    $validCsharpContent = ((Get-Content -Path $CsharpFilePath) | ForEach-Object {($_ -replace "\s+", "")} `
+        | Where-Object {(($_ -notlike "*//*") -and ($_ -ne "") -and ($_ -ne '{') -and ($_ -ne '}') -and ($_ -ne '};'))})
+
+    $validPowerShellContent = ((Get-Content -Path $PowerShellFilePath)| ForEach-Object {($_ -replace "\s+", "")} `
+        | Where-Object {(($_ -notlike "*//*") -and ($_ -ne "") -and ($_ -ne '{') -and ($_ -ne '}') -and ($_ -ne '};'))})
+    
+    if ($validCsharpContent.Length -ne $validPowerShellContent.Length) {
+        return @{"compared" = $false; "result" = "$fileName - content different"}
+    }
+    foreach($csharpContent in $validCsharpContent) {
+        if ($validPowerShellContent -notcontains $csharpContent)
+        {
+            return @{"compared" = $false; "result" = "$fileName - content different"}
+        }
+    }
+
+    return @{"compared" = $true; "result" = $null}
+
+}
+
+function CompareTestSdkCode { 
+    param (
+        [TestSdkModel]
+        $TestSdk,
+        
+        [string[]]
+        $ignoreFiles
+    )
+    if (($TestSdk.PowerShellGeneratedStatus -eq [GeneratedStatus]::Failed) -or ($TestSdk.CsharpGeneratedStatus -eq [GeneratedStatus]::Failed)) {
+        $TestSdk.SdkComparedStatus = [ComparedStatus]::NotEqual
+        return
+    } else {
+        $csharpFiles = Get-ChildItem -Path $TestSdk.CsharpSdkFolder -Recurse -File -Exclude $ignoreFiles
+        $powershellFiles = Get-ChildItem -Path $TestSdk.PowerShellSdkFolder -Recurse -File -Exclude $ignoreFiles
+        $compareResult = (CompareSdkGenerateFileCount -CsharpSdkFileNames $csharpFiles.Name -PowerShellSdkFileNames $powershellFiles.Name)
+        $isFileContentEqual = $true
+        if ($null -ne $compareResult["result"]) {
+            $TestSdk.Result +=  $compareResult["result"]
+            $isFileContentEqual = $false
+        }
+        # Get files that exist in csharp and powershell
+        $existCsharpFiles = $csharpFiles | Where-Object {$compareResult["files"] -contains $_.Name}
+
+        foreach($existFile in $existCsharpFiles)
+        {
+            $powershellFile = $powershellFiles | Where-Object {$_.Name -eq $existFile.Name}
+            $fileCompareResult = CompareSingleSdkFile -CsharpFilePath $existFile.FullName -PowerShellFilePath $powershellFile.FullName
+            if (!$fileCompareResult["compared"]) {
+                $TestSdk.Result +=  $fileCompareResult["result"]
+                $isFileContentEqual = $false
+            }
+        }
+
+        if ($isFileContentEqual) {
+            $TestSdk.SdkComparedStatus = [ComparedStatus]::Equal
+        } else {
+            $TestSdk.SdkComparedStatus = [ComparedStatus]::NotEqual
+        }
+    } 
+}
+
+try 
+{
+    $script:autorestCsharp = Join-Path $PSScriptRoot "..\..\..\..\autorest.csharp"
```

Comment:
Make $script:autorestCsharp, $script:autorestPowerShell as two optional parameters, and current value as default values

Reply:
Done